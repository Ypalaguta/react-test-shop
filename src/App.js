import { React, useState } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom'

import GlobalStyles from './components/global-styles/global-styles.style'
import Header from './components/header/header.component'

import MainPage from './pages/main/main.component'
import CategoriesPage from './pages/categories/categories.component'
import ContactsPage from './pages/contacts/contacts.component'
import AuthPage from './pages/auth/auth.component'
import CheckoutPage from './pages/checkout/checkout.component'

function App() {
  const [userInfo, getUserInfo] = useState(null);
  return (
    <div className="App">
      <GlobalStyles />
      <Header />
      <Switch>
        <Route path="/categories" component={CategoriesPage} />
        <Route path="/contacts" component={ContactsPage} />
        <Route path="/auth">
          {userInfo ? <Redirect to="/" /> : <AuthPage />}
        </Route>
        <Route path="/checkout" component={CheckoutPage} />
        <Route path="/" component={MainPage} />
      </Switch>
    </div>
  );
}

export default App;
