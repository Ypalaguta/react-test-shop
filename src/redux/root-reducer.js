import {combineReducers} from "redux";
import {persistReducer} from "redux-persist";
import storage from 'redux-persist/lib/storage';

import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";
import mainReducer from "./main/main.reducer";
import categoriesReducer from "./categories/categories.reducer";

const persistConfig = {
    key: 'root',
    storage,
    whitelist: ['cart']
}

const rootReducer = combineReducers({
    user: userReducer,
    cart: cartReducer,
    main: mainReducer,
    categories: categoriesReducer,
});

export default persistReducer(persistConfig, rootReducer);