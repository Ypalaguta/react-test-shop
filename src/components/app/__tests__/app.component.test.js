import React from 'react';
import {shallow} from "enzyme";
import {Provider} from 'react-redux';

import App from '../app.component'
import {store} from '../../../redux/store'

test('app component render snapshot', ()=>{
    const wrapper = shallow(<Provider store={store}> test </Provider>);
    // const wrapper = shallow(<Provider store={store}> <App/> </Provider>);
    expect(wrapper).toMatchSnapshot();
})