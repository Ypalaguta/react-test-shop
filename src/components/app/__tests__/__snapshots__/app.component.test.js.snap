// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component render snapshot 1`] = `ShallowWrapper {}`;
